### ISSUE TITLE
Controlling optimizer.step()

### ISSUE THREAD

**Issue by marcmk6** [pytorch/fairseq#3715]
## ❓ Questions and Help
Hi,

I'm dealing with a very special loss which requires significant amount of memory to calculate. Unfortunately I don't have GPUs with enough memory. So I decide to calculate the first part of the loss, perform loss.backward() and then calculate the second part, loss.backward() and finally optimizer.step(). 

I'm wondering where should I modify so that optimizer.step() is performed after more than one back propagation?

Thanks
